src/20240119_Gho_GHOStabilityModule/Gho_GHOStabilityModule_20240119.sol

```diff
@@ -0,0 +1,124 @@
+ * 2. GHO Stability Module Update
+ *  - Discussion: https://governance.aave.com/t/gho-stability-module-update/14442
+ * 3. GHO Stability Module Launch
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0xe9b62e197a98832da7d1231442b5960588747f184415fba4699b6325d7618842
+ */
+contract Gho_GHOStabilityModule_20240119 is IProposalGenericExecutor {
+  using SafeERC20 for IERC20;
+
+  address public constant GSM_USDC = 0x0d8eFfC11dF3F229AA1EA0509BC9DFa632A13578;
+  address public constant GSM_USDC_ORACLE_SWAP_FREEZER = 0xD9096444807Da3D05EcA6d1E19380133A59394A6;
+  address public constant GSM_USDT = 0x686F8D21520f4ecEc7ba577be08354F4d1EB8262;
+  address public constant GSM_USDT_ORACLE_SWAP_FREEZER = 0x0F1773be3CaA314273A69dfE1A107814893C359F;
+  address public constant GSM_REGISTRY = 0x167527DB01325408696326e3580cd8e55D99Dc1A;
+  address public constant GSM_FIXED_FEE_STRATEGY = 0xD4478A76aCeA81D3768A0ACB6e38f25eEB6Eb1B5;
+
+  string public constant GSM_USDC_FACILITATOR_LABEL = 'GSM USDC';
+  uint128 public constant GSM_USDC_BUCKET_CAPACITY = 500_000e18;
+  string public constant GSM_USDT_FACILITATOR_LABEL = 'GSM USDT';
+  uint128 public constant GSM_USDT_BUCKET_CAPACITY = 500_000e18;
+
+  address public constant ROBOT_OPERATOR = 0x020E452b463568f55BAc6Dc5aFC8F0B62Ea5f0f3;
+  uint96 public constant LINK_AMOUNT_ORACLE_FREEZER_KEEPER = 10 ether;
+  uint96 public constant TOTAL_LINK_AMOUNT_KEEPERS = LINK_AMOUNT_ORACLE_FREEZER_KEEPER * 2; // 2 GSMs
+
+  function execute() external {
+    // 1. Enroll GSMs as GHO Facilitators
+    IGhoToken(AaveV3Ethereum.GHO_TOKEN).addFacilitator(
+      GSM_USDC,
+      GSM_USDC_FACILITATOR_LABEL,
+      GSM_USDC_BUCKET_CAPACITY
+    );
+    IGhoToken(AaveV3Ethereum.GHO_TOKEN).addFacilitator(
+      GSM_USDT,
+      GSM_USDT_FACILITATOR_LABEL,
+      GSM_USDT_BUCKET_CAPACITY
+    );
+
+    // 2. Add GSM Swap Freezer role to OracleSwapFreezers
+    IGsm(GSM_USDC).grantRole(IGsm(GSM_USDC).SWAP_FREEZER_ROLE(), GSM_USDC_ORACLE_SWAP_FREEZER);
+    IGsm(GSM_USDT).grantRole(IGsm(GSM_USDT).SWAP_FREEZER_ROLE(), GSM_USDT_ORACLE_SWAP_FREEZER);
+
+    // 3. Update Fee Strategy
+    IGsm(GSM_USDC).updateFeeStrategy(GSM_FIXED_FEE_STRATEGY);
+    IGsm(GSM_USDT).updateFeeStrategy(GSM_FIXED_FEE_STRATEGY);
+
+    // 4. Add GSMs to GSM Registry
+    IGsmRegistry(GSM_REGISTRY).addGsm(GSM_USDC);
+    IGsmRegistry(GSM_REGISTRY).addGsm(GSM_USDT);
+
+    // 5. Register OracleSwapFreezer as keepers
+    AaveV3Ethereum.COLLECTOR.transfer(
+      AaveV3EthereumAssets.LINK_UNDERLYING,
+      address(this),
+      TOTAL_LINK_AMOUNT_KEEPERS
+    );
+    IERC20(AaveV3EthereumAssets.LINK_UNDERLYING).forceApprove(
+      ROBOT_OPERATOR,
+      TOTAL_LINK_AMOUNT_KEEPERS
+    );
+
+    IAaveCLRobotOperator(ROBOT_OPERATOR).register(
+      'GSM USDC OracleSwapFreezer',
+      GSM_USDC_ORACLE_SWAP_FREEZER,
+      5000000, // gasLimit
+      LINK_AMOUNT_ORACLE_FREEZER_KEEPER
```

Comment:
Funding the keeper with 10 Link would not be sufficient for gasLimit of 5m as the keeper will then be underfunded and would not perform any action (this is unless you wish to fund it yourself later and do not wish for the DAO to pay)

Reply:
For reference, with the current gasPrice of 23 gwei and gasLimit of 5m the keeper requires a minimum link balance of 42 LINK

Reply:
Adjusted the gasLimit to 150k, which is the double of the gas cost for the keep action, and increased the LINK amount to 100 ether. Based on your example, we would have enough LINK for almost 80 keep actions on each

Reply:
decreased to 80 LINK each, so there is enough funds in the Treasury to pull